tests/writing.rst

```diff
@@ -15,13 +37,48 @@ top), a table of tests (at the bottom), and a tiny main which sets the
 test harness in motion. It uses the table to figure out what the tests
 are called, and what function to call to run them.
 
+Incidentally, when fixing bugs, it's often better to write the test
+before fixing the bug. Firstly, it's easier to assure yourself that the
+bug is (a) genuine, and (b) fixed, because you see the test go from fail
+to pass (though sometimes you don't get the testcase quite right, so
+this isn't doesn't always work as well as it should). Secondly you're
+more likely to write the test carefully, because once you've fixed
+something there's often a feeling that you should commit it for the good
+of the world, which tends to distract you.
+
+The framework is done for you, so you don't need to worry about that
+much. You are responsible for doing two things:
+
+* writing a minimal test or tests for the feature
+* adding that test to the list of tests to be run
+
+Adding the test is simple. There's a ``test_desc`` array in each file that
+comprises a set of tests (I'll come to that in a minute), and you just
+add another entry. The entry is an array consisting of a name for the
+test and a pointer to the function that is the test. Easy. The procedure
+is even simpler for apitest tests - there you just use ``DEFINE_TESTCASE``
+to define your new testcase, and a script picks it up and makes sure it
+is run.
```

Comment:
I think it's probably better to talk about the easy `apitest` situation first, since that's where most new tests go.

Reply:
(Ah, I see this was existing text just moved around...)